### PR TITLE
Use asynchronous writers for discovery

### PR DESCRIPTION
**Pull request by ivanpauno**
For really small `message_size_max`, you also need to set the discovery writers to asynchronous publish mode.

Maybe it's also possible to set different transport settings for discovery and data, I will double check that.

### CODE REVIEW

File: etc/ROS2TEST_QOS_PROFILES.xml [hunk 1]
```diff
@@ -28,6 +28,20 @@ xsi:noNamespaceSchemaLocation="http://community.rti.com/schema/5.3.1/rti_dds_qos
...
+        <discovery_config>
+          <publication_writer_publish_mode>
+            <kind>ASYNCHRONOUS_PUBLISH_MODE_QOS</kind>
+          </publication_writer_publish_mode>
+          <subscription_writer_publish_mode>
+            <kind>ASYNCHRONOUS_PUBLISH_MODE_QOS</kind>
+          </subscription_writer_publish_mode>
+          <!--secure_volatile_writer_publish_mode>
+            <kind>ASYNCHRONOUS_PUBLISH_MODE_QOS</kind>
+          </secure_volatile_writer_publish_mode-->
```
Comment (ivanpauno) [inline etc/ROS2TEST_QOS_PROFILES.xml:40]: I think this is only a connexy 6 thing, I'm not completely sure.

File: package.xml [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
   <buildtool_depend>ament_cmake</buildtool_depend>
 
-  <exec_depend>gazebo9</exec_depend>
+  <exec_depend>gazebo11</exec_depend>
```
Comment (ivanpauno) [inline package.xml:12]: I'm using gazebo11 locally, it doesn't matter here though

**Comment by ivanpauno** [comment]
Docs about the qos being set can be found https://community.rti.com/static/documentation/connext-dds/5.3.1/doc/api/connext_dds/api_cpp/structDDS__DiscoveryConfigQosPolicy.html, https://community.rti.com/static/documentation/connext-dds/5.3.0/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/Content/UsersManual/DISCOVERY_CONFIG_Qos.htm.

File: etc/ROS2TEST_QOS_PROFILES.xml (hunk 1 above)
Comment (ivanpauno) [reply etc/ROS2TEST_QOS_PROFILES.xml:40]: Yes, I'm pretty sure this is connext 6 only.

**Review by jacobperron — APPROVED** [review]

**Pull request merged**